packages/mangojuice-core/__tests__/block/Computed.spec.js

```diff
@@ -91,56 +85,46 @@ describe("Computed", () => {
   describe('Computed with dependencies', () => {
     const SharedBlock = {
       createModel: () => ({ e: 0, f: 4, g: 6 }),
-      Logic: {
-        name: "SharedBlock",
+      Logic: class SharedBlock {
         computed() {
-          const { model } = this;
           return {
-            e: () => model.f + model.g
+            e: () => this.model.f + this.model.g
           };
-        },
-        @Cmd.update
-        SetField(name, value) {
+        }
+        @cmd SetField(name, value) {
           return { [name]: value };
         }
       }
     };
     const AppBlock = {
       createModel: () => ({ a: 1, b: 2, c: 0, d: 0 }),
-      Logic: {
-        name: "AppBlock",
+      Logic: class AppBlock {
         computed() {
-          const { model, shared, depends } = this;
+          const { model, shared } = this;
           return {
             c: depends(shared).compute(() => shared.f + shared.e + model.a),
             d: depends(shared).compute(() => shared.f + shared.g + model.b)
           };
-        },
-        @Cmd.update
-        SetField(name, value) {
+        }
+        @cmd SetField(name, value) {
           return { [name]: value };
         }
       }
     };
     const AppParent = {
       createModel: () => ({ a: 1, c: 0, d: 0, child: AppBlock.createModel() }),
-      Logic: {
-        name: "AppParent",
+      Logic: class AppParent {
         children() {
-          const { nest } = this;
-          return {
-            child: nest(AppBlock.Logic)
-          };
-        },
+          return { child: child(AppBlock.Logic) };
+        }
         computed() {
-          const { model, shared, depends } = this;
+          const { model, shared } = this;
           return {
             c: depends(shared, model.child).compute(() => shared.f + shared.e + model.a + model.child.a),
```

Comment:
How will it be called? Shared or model.child change calls computed() entirely?

Reply:
Not really. Will be called only compute function of field “c”